app/scripts/metamask-controller.js

```diff
@@ -2294,6 +2310,7 @@ export default class MetamaskController extends EventEmitter {
     this.accountTracker.stop();
     this.txController.stopIncomingTransactionPolling();
     this.tokenDetectionController.disable();
+    this.nftDetectionController.stop();
```

Comment:
Why is the controller started/stopped in both `triggerNetworkrequests/stopNetworkRequests` and KeyringController controller lock/unlock?  Why is either one not sufficient on its own?

Reply:
Locking MM does not trigger "controllerConnectionChanged" event, which leaves MM still making requests in the background when the user clicks on "Lock Metamask" option in menu

Reply:
I think we could live with that, since they still technically have MM open.

Seems safer to follow the pattern of other controllers.  I'd be a little worried that the KeyringController events could fire at times we don't expect.  And about double starts/stops if it happens in both places.  What do you think?

Reply:
Hmmmm i still dnt see how this can cause an issue but I guess we can be conservative and remove this from lock/unlock. Do not have the numbers of how many users lock their MM but we can assume its not that much x))